File: app/src/main/java/com/ivanmagda/habito/barchart/habitoBarChartRange.java
package com.ivanmagda.habito.barchart;

import android.content.Context;
import android.content.res.Resources;
import android.support.annotation.NonNull;

import com.ivanmagda.habito.R;

import java.util.Arrays;
import java.util.List;

public final class HabitoBarChartRange {

    public enum DateRange {
        WEEK, MONTH, YEAR;

        public String stringValue(@NonNull final Context context) {
            Resources resources = context.getResources();
            switch (this) {
                case WEEK:
                    return resources.getString(R.string.week);
                case MONTH:
                    return resources.getString(R.string.month);
                case YEAR:
                    return resources.getString(R.string.year);
                default:
                    throw new IllegalArgumentException("Receive illegal date range string");
            }
        }

        public static DateRange fromString(@NonNull String string, @NonNull final Context context) {
            if (string.equals(WEEK.stringValue(context))) {
                return WEEK;
            } else if (string.equals(MONTH.stringValue(context))) {
                return MONTH;
            } else if (string.equals(YEAR.stringValue(context))) {
                return YEAR;
            } else {
                throw new IllegalArgumentException("Illegal date string");
            }
        }
    }

    public static List<String> allStringValues(@NonNull final Context context) {
        return Arrays.asList(DateRange.WEEK.stringValue(context),
                DateRange.MONTH.stringValue(context),
                DateRange.YEAR.stringValue(context));
    }

    private HabitoBarChartRange() {
    }

}

File: app/src/main/java/com/ivanmagda/habito/barchart/HabitoBarChartRange.kt
package com.ivanmagda.habito.barchart

import android.content.Context
import com.ivanmagda.habito.R
import java.util.*

object HabitoBarChartRange {

    enum class DateRange {
        WEEK, MONTH, YEAR;

        fun stringValue(context: Context): String {
            val resources = context.resources
            return when (this) {
                WEEK -> resources.getString(R.string.week)
                MONTH -> resources.getString(R.string.month)
                YEAR -> resources.getString(R.string.year)
            }
        }

        companion object {
            fun fromString(string: String, context: Context): DateRange {
                return when (string) {
                    WEEK.stringValue(context) -> WEEK
                    MONTH.stringValue(context) -> MONTH
                    YEAR.stringValue(context) -> YEAR
                    else -> throw IllegalArgumentException("Illegal date string")
                }
            }
        }
    }

    fun allStringValues(context: Context): List<String> {
        return Arrays.asList(DateRange.WEEK.stringValue(context), DateRange.MONTH.stringValue(context), DateRange.YEAR.stringValue(context))
    }
}
